### PR TITLE
Add travis-ci markdown linter

### PR DESCRIPTION
**Pull request by hgibson1**
Runs remarkjs

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+  - "node"
+before_script:
+  npm install -g express remark-lint remark-cli remark-preset-lint-consistent remark-preset-lint-markdown-style-guide remark-preset-lint-recommended
+script: remark --rc-path $TRAVIS_BUILD_DIR/.remarkrc $TRAVIS_BUILD_DIR/docs/source
```
Comment (larsks) [inline .travis.yml]: Should we run both the link checker and the linter as part of the travis configuration?

Comment (hgibson1) [reply .travis.yml]: Sure. It might make sense to run the sphinx build also.

File: .travis.yml [hunk 2]
```diff
@@ -12,3 +12,9 @@ matrix:
       install:
         pip install -r $TRAVIS_BUILD_DIR/requirements-testing.txt
       script: tox -c $TRAVIS_BUILD_DIR
+    - language: python
+      python:
+        - 2.7
+      install:
+        pip install -r $TRAVIS_BUILD_DIR/requirements-testing.txt
```
Comment (larsks) [inline .travis.yml]: Shouldn't this actually be `$TRAVIS_BUILD_DIR/requirements.txt`? The sphinx build doesn't need `requirements-testing.txt`.

File: requirements-testing.txt [hunk 3]
```diff
@@ -1,3 +1,6 @@
 click
 mistune
 tox
+sphinx
```
Comment (larsks) [inline requirements-testing.txt]: I would just keep these in `requirements.txt`, since they are required to actually build the docs.

**Pull request merged**